packages/core/src/editor/BlockNoteExtensions.ts

```diff
@@ -159,13 +165,45 @@ const getTipTapExtensions = <
     // marks:
     Link.extend({
       inclusive: false,
-      addKeyboardShortcuts() {
-        return {
-          "Mod-k": () => {
-            this.editor.commands.toggleLink({ href: "" });
-            return true;
-          },
-        };
+      // Adapted from https://github.com/ueberdosis/tiptap/blob/a0d2f2803652851bbe2f06f124a70bc01cfb0dab/packages/extension-link/src/link.ts#L301
+      // Fixes hrefs without a protocol prefix. Normally, these are treated as
+      // relative URLs, which doesn't make much sense for users. So if href is
+      // e.g. "www.google.com", it will now redirect users to:
+      // "https://www.google.com"
+      // instead of
+      // "<protocol>://<current-domain>/www.google.com".
+      renderHTML({ HTMLAttributes }) {
+        // prevent XSS attacks
+        if (!isAllowedUri(HTMLAttributes.href, this.options.protocols)) {
+          // strip out the href
+          return [
+            "a",
+            mergeAttributes(this.options.HTMLAttributes, {
+              ...HTMLAttributes,
+              href: "",
+            }),
+            0,
+          ];
+        }
+
+        // Modified section
+        // If the href does not start with a protocol, prefix with default one.
+        let prefix = this.options.defaultProtocol + "://";
```

Comment:
I think this makes sense, but only fixes it at the prosemirror level "when rendering". This means in the blocknote schema, it will still show the URL without href.

I think we should also fix the input validation, so that we add the "https://" prefix there, if there was no scheme entered

Reply:
R.e. the first part I think that's what we want. I originally implemented it the other way, so `https://` was prepended to the URL when it's set, and saving that to the URL prop. But I slightly preferred the UX with this approach, as if you add a link to e.g. `www.google.com` and then edit it, you'll see the same link you added instead of  `https://www.google.com`. Imo this is better for users since the overall link is shorter and more legible + more intuitive for the user if BlockNote doesn't modify the links that users set.

Reply:
Hmm, the downside I see is that when the consumer then processes blocknote json (for whatever reason), they need to apply the same logic to the links.

Notion (and I think google docs as well) automatically add the https to the ux. While yes, you then edit something differently then you inserted, it makes it very explicit what the actual link is